lib/bolt/analytics.rb

```diff
@@ -0,0 +1,141 @@
+require 'bolt/version'
+require 'httpclient'
+require 'json'
+require 'locale'
+require 'logging'
+require 'securerandom'
+
+module Bolt
+  module Analytics
+    ProtocolVersion = 1
+    ApplicationName = 'bolt'
+    TrackingID = 'UA-fakefakefake-1'
+    TrackingURL = 'https://google-analytics.com/collect'
+
+    def self.build_client
+      logger = Logging.logger[self]
+
+      config_file = File.expand_path('~/.puppetlabs/bolt/analytics.yaml')
+      config = load_config(config_file)
```

Comment:
doesn't this need to be load_config(nil, [config_file]) to avoid erroring when the file is absent

Reply:
This is using a `load_config` method from the Analytics class, not the standard `read_config_file` util method. The reason is mostly that we don't want to swallow exceptions. If the config file *doesn't exist*, this `load_config` method will return a default empty config. But if the config file can't be read for some *other* reason (permissions, bad YAML, etc) it will be handled by the `rescue` in this method, which will err on the side of caution and disable analytics.